### PR TITLE
[MOE] add unitest for MOE experts layout, gradient handler and kernel

### PR DESCRIPTION
**Pull request by 1SAA**
- add unitest for MOE experts layout, gradient handler and kernel
- remove old MOE initialization

### CODE REVIEW

File: tests/test_moe/test_top1.py [hunk 1]
```diff
@@ -7,82 +7,81 @@
...
+    torch.manual_seed(rs + local_rank)    # set each process has different random seed
+
+    # get randomized data
     tokens = torch.randn(BATCH_SIZE, hidden_size, dtype=data_type, device=get_current_device(), requires_grad=True)
-    # print(f"tokens:\n{tokens}")
-    router = Top1Router(1)
-    layer = MoeLayer(hidden_size, NUM_EXPERTS, router, nn.Identity())
+
+    expert = Experts(nn.Identity, NUM_EXPERTS)
+    layer = MoeLayer(hidden_size, NUM_EXPERTS, Top1Router(capacity_factor_train=1.0), expert)
```
Comment (feifeibear) [inline tests/test_moe/test_top1.py:36]: why not merge test_top1.py with test_top2.py?
The only diff is the parameters Top1Router.

File: tests/test_moe/test_kernel.py [hunk 2]
```diff
@@ -8,76 +8,82 @@
...
+    expert = Experts(expert_module, NUM_EXPERTS, **expert_factor)
+    layer = MoeLayer(hidden_size, NUM_EXPERTS, Top2Router(capacity_factor_train=1.0), expert)
     if data_type == torch.float16:
         layer = layer.half()
-    layer.cuda_mode = False
-
-    old_out = layer(tokens)
 
+    layer.use_kernel = False
+    old_out = layer(tokens)    # get outputs through matrix multiplication
```
[the remark this answers is not in the document]
Comment (1SAA) [reply tests/test_moe/test_kernel.py:43]: use MM to substitute cuda kernel

File: tests/test_moe/test_grad_handler.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+    MOE_CONTEXT.setup(42)    # MOE initialization
+    noisy_func = UniformNoiseGenerator()
+    router = Top1Router(noisy_func=noisy_func)
+    exp0 = Experts(expert_module, 1, **expert_factor)
+    layer1 = MoeLayer(DIM, 1, router, exp0)
+    exp1 = Experts(expert_module, 2, **expert_factor)
+    layer2 = MoeLayer(DIM, 2, router, exp1)
+    exp2 = Experts(expert_module, 4, **expert_factor)
+    layer3 = MoeLayer(DIM, 4, router, exp2)
```
[the remark this answers is not in the document]
Comment (1SAA) [reply tests/test_moe/test_grad_handler.py]: done

File: tests/test_moe/test_kernel.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (1SAA) [reply tests/test_moe/test_kernel.py]: done

**Review by feifeibear — APPROVED** [review]
